packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -397,7 +397,7 @@ export default function transformClass(
       return ref;
     };
 
-    for (const bareSuper of classState.bareSupers) {
+    for (const bareSuper of new Set(classState.bareSupers)) {
```

Comment:
Is this `new Set` to avoid duplicated paths? When can it happen? I think that you should add a comment.

Reply:
```
super[super().name]();
```

We find the original `super`, and transform it. But the MemberExpression `super[NEXT]` is still queued, so we find `super()` and add it. Then the transformed `_get(PROTO, super().name)` is traversed, finding `super()` again.

Reply:
Maybe the set would be better inside ReplaceSupers?

Reply:
Yeah, state.bareSupers should be a set since the beginning if possible.

Reply:
Done.

Reply:
I think that you forgot to commit the file? In the last commit only some tests and comments are changed.

Reply:
Whoops.